internal/dbmodel/role.go

```diff
@@ -0,0 +1,44 @@
+// Copyright 2024 Canonical.
+
+package dbmodel
+
+import (
+	"time"
+
+	"github.com/juju/names/v5"
+
+	jimmnames "github.com/canonical/jimm/v3/pkg/names"
+)
+
+// RoleEntry holds a role entry.
+type RoleEntry struct {
+	// Note this doesn't use the standard gorm.Model to avoid soft-deletes.
+	ID        uint `gorm:"primarykey"`
+	CreatedAt time.Time
```

Comment:
does it mean we need to set these fields manually?

Remember to handle it next PR